### PR TITLE
[TrimmableTypeMap] Scanner and JCW edge-case fixes

### PR DESCRIPTION
**Pull request by simonrozsival**
## Summary

Fix scanner and JCW generation edge cases in the trimmable type map path.

This PR is now limited to the scanner/JCW fixes and associated regression tests.

### Changes

- **Override detection across intermediate MCW bases**
  - Continue walking base types when looking for inherited [Register] methods and properties, even when an intermediate DoNotGenerateAcw base sits between the concrete ACW and the original declaration.
  - Covers both normal and generic-instantiated base chains.

- **Deferred registerNatives() for Application / Instrumentation types**
  - Generate lazy __md_registerNatives() calls for types that cannot safely register from a static initializer.
  - Keeps marshal methods wired up without eager static registration.

- **Proxy name sanitization cleanup**
  - Sanitize generated proxy type names for generic managed types by replacing namespace / nesting / generic-arity separators and appending _Proxy in a dedicated ModelBuilder helper.

- **Regression tests**
  - Add fixture hierarchies and unit coverage for:
    - override discovery through intermediate MCW bases
    - override discovery through generic intermediate bases
    - deferred registration behavior for Application / Instrumentation
    - CRC64 naming expectations for unregistered types

### CODE REVIEW

File: tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/JcwJavaSourceGeneratorTests.cs [hunk 1]
```diff
@@ -257,6 +261,61 @@ public void Generate_MarshalMethod_HasOverrideAndNativeDeclaration ()
...
+						JniSignature = "()V",
+						ManagedMethodName = "OnStart",
+						NativeCallbackName = "n_OnStart",
+						Connector = "GetOnStartHandler",
+					},
+				},
+			};
+
+			var java = GenerateToString (type);
+			AssertContainsLine ("__md_registerNatives ();\n\t\tn_OnCreate_Landroid_os_Bundle_ (p0);\n", java);
```
Comment (simonrozsival) [inline tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/JcwJavaSourceGeneratorTests.cs:315]: 🤖 💡 **Testing** — This nicely locks in the per-callback `__md_registerNatives()` fallback. Since `CannotRegisterInStaticConstructor` still also depends on `ApplicationRegistration.registerApplications()` for the startup path, consider adding one regression test for `result.ApplicationRegistrationTypes` or the generated `ApplicationRegistration.java` as well. That would keep both halves of the deferred-registration contract covered if this generator gets refactored again.

_Rule: Add regression coverage for the full bug-fix path._

**Review by simonrozsival — COMMENTED** [review]
## 🤖 AI Review Summary

**Verdict**: ⚠️ Needs Changes

I didn’t spot a correctness bug in the scanner / JCW fixes themselves, but I did leave one targeted regression-coverage suggestion below. Public checks are green, however the internal `Xamarin.Android-PR` pipeline is still running, so this PR is not mergeable yet.

- 💡 **Testing**: cover the `ApplicationRegistration.registerApplications()` half of the deferred-registration flow too (`tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/JcwJavaSourceGeneratorTests.cs:315`).

Nice cleanup overall — the override walk past intermediate MCW bases and the generic proxy-name sanitization both make sense.

---
_Review generated by android-reviewer from [review guidelines](https://github.com/dotnet/android/blob/main/.github/skills/android-reviewer/references/review-rules.md)._
